### PR TITLE
Feature/df 1425 os pilot activation page

### PR DESCRIPTION
**Pull request by 99999989**
Successfully tested on staging. 
Here are some valid codes for staging:
```
P7BOXAPK
7AA5PZFB
LBNKWGX4
D7C6NL2O
3JTG2QAB
```

### CODE REVIEW

File: alv-portal-ui/src/app/candidate/candidate-search/candidate-detail/candidate-detail-model-factory.spec.ts [hunk 1]
```diff
@@ -74,6 +75,9 @@ describe('CandidateDetailModelFactory', () => {
     },
     isLegalTermAcceptanceRequired(): boolean {
       return false;
+    },
+    hasPilotFeature(featureCode: FeatureCode): boolean {
```
Comment (pavax) [inline alv-portal-ui/src/app/candidate/candidate-search/candidate-detail/candidate-detail-model-factory.spec.ts]: hasFeature

File: alv-portal-ui/src/app/shared/layout/main-navigation/menu-entry.service.ts [hunk 2]
```diff
@@ -125,14 +127,14 @@ const ONLINE_FORMS_MENU_ENTRIES: Array<MenuEntry> = [
     iconKey: IconKey.WORK_EFFORTS,
     labelKey: 'portal.navigation.menu-entry.work-efforts',
     path: ['work-efforts'],
-    userPredicate: (u) => isAuthenticatedUser(u)
+    userPredicate: (u) => hasPilotFeature(u, FeatureCode.NPA)
```
Comment (pavax) [inline alv-portal-ui/src/app/shared/layout/main-navigation/menu-entry.service.ts]: please remove also the 'old' check for the profile info (for example staging)

File: alv-portal-ui/src/app/core/auth/has-pilot-feature-guard.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  providedIn: 'root'
+})
+export class HasPilotFeatureGuard extends AbstractAuthenticationGuard {
+
+  constructor(authenticationService: AuthenticationService, landingNavigationService: LandingNavigationService) {
+    super(authenticationService, landingNavigationService);
+  }
+
+  protected canUserActivate(user: User, route: ActivatedRouteSnapshot): boolean {
+    return hasPilotFeature(user, route.data.featureCode);
```
Comment (pavax) [inline alv-portal-ui/src/app/core/auth/has-pilot-feature-guard.service.ts]: use variable featureName

File: alv-portal-ui/src/app/shared/backend-services/feature-code-list/feature-code-list.types.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+
+export enum FeatureCode {
```
Comment (pavax) [inline alv-portal-ui/src/app/shared/backend-services/feature-code-list/feature-code-list.types.ts]: rename to FeatureName

File: alv-portal-ui/src/app/candidate/candidate-search/candidate-detail/candidate-detail-model-factory.spec.ts (hunk 1 above)
Comment (99999989) [reply alv-portal-ui/src/app/candidate/candidate-search/candidate-detail/candidate-detail-model-factory.spec.ts]: done

File: alv-portal-ui/src/app/shared/layout/main-navigation/menu-entry.service.ts (hunk 2 above)
Comment (99999989) [reply alv-portal-ui/src/app/shared/layout/main-navigation/menu-entry.service.ts]: Ok, removed

**Comment by emina-str** [comment]
3JTG2QAB taken.

> Successfully tested on staging.
> Here are some valid codes for staging:
> 
> ```
> P7BOXAPK
> 7AA5PZFB
> LBNKWGX4
> D7C6NL2O
> 3JTG2QAB
> ```

**Pull request merged**